xml/templates/dao.tpl

```diff
@@ -18,12 +18,22 @@ class {$table.className} extends CRM_Core_DAO {ldelim}
      {if !empty($table.component)}const COMPONENT = '{$table.component}';{/if}
 
      /**
-      * Static instance to hold the table name.
+      * Name of SQL table for the {$table.entity} entity.
       *
       * @var string
       */
       public static $_tableName = '{$table.name}';
 
+   {* Only print this variable if it's different than the default in CRM_Core_DAO *}
+   {if count($table.primary_key) !== 1 || $table.primary_key.0 !== 'id'}
+     /**
+      * Primary key field(s).
+      *
+      * @var string[]
+      */
+      public static $_primaryKey = [{if $table.primary_key}'{"', '"|implode:$table.primary_key}'{/if}];
```

Comment:
Changing this to implode:$table.primaryKey.field} seems to have the desired effect


![image](https://user-images.githubusercontent.com/336308/124371090-7fc82c80-dcd2-11eb-8b27-0224301647f4.png)